config/env_spec_ext.go

```diff
@@ -96,15 +109,25 @@ func NewEnvironmentSpecExt(spec *EnvironmentSpec) (*EnvironmentSpecExt, error) {
 		}
 
 		for i := range api.Operations {
+			isGRPC := api.GrpcService != ""
 			op := api.Operations[i]
 
-			if len(op.HTTPMatches) == 0 { // empty is wildcard
-				split = []string{api.ID, wildcard, wildcard}
+			// For gRPC APIs, always interpret the op Name as a gRPC method, so add it as a
+			// child of the API.
+
+			if isGRPC {
+				split := []string{api.ID, "POST", "", op.Name}
```

Comment:
Why is there an empty string between `"POST"` and `op.Name`?

Reply:
Right, empty strings are skipped in the tree and during matching. But if for some reason it is necessary here, we should add a comment explaining why.

Reply:
Right, empty strings are skipped in the tree and during matching. But if for some reason it is necessary here, we should add a comment explaining why.